### PR TITLE
genre support for movies

### PR DESCRIPTION
**Pull request by cbarbara**
I added this locally to make it faster for me to build channels, but figured I might as well put up a PR if you are interested.
I took a look at how you added collection support for movies and tried something similar for genres.

Screen Shot 2020-08-16 at 12 18 31 PM
Screen Shot 2020-08-16 at 12 18 54 PM

Genres for TV shows are also supported by plex but the UI did not support that extra level of nesting, which is why its explicitly only added for movies. I also had an earlier version that put Genre's as a top element like Playlists, but ended up here.

Long story short, feel free to reject if this is not something you think is needed.

### CODE REVIEW

**Comment by vexorian** [comment]
I wish there was somewhere I could write this, but all PRs should go to `dev/0.0.x . So please change the branch.

Regarding the feature. It was not something I wanted to implement because in my opinion, it makes more sense to leave that job to Plex. And for example if you wanted to grab all Action movies in your library, I thought the better way to do it would be to create a smart playlist and then import that play list. 

My other concern is that the drop down is already quite full of stuff , specially now that I've added collections. But my real fear is that people will then start requesting drop downs for other things, we would have to draw a line somewhere.

But it  certainly is something that people request, and at least in the case of genres, it makes sense that people would want to be able to make genre-specific channels quickly. I think I'll go around and give it a test and think about it. But first, change the branch.

**Comment by cbarbara** [comment]
@vexorian - updated.

**Comment by cbarbara** [comment]
> I wish there was somewhere I could write this, but all PRs should go to `dev/0.0.x`

You can use a pull request template to provide that kind of info when someone is creating the PR - https://docs.github.com/en/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository 
https://github.com/stevemao/github-issue-templates shows many examples.

**Comment by vexorian** [comment]
Ok I am going to add it. It's interesting to see the genres in action.

Just one thing I've noticed when testing it. When you click the + sign on the library, it will add the moveis that are inside the genres multiple times, because the library adder is also adding everything in each genre. Like collections, something must be in place to prevent this from happening. I can fix this tonight or tomorrow but if you have the time, please give it a try before I merge it.

**Comment by MoRBiD-MaNGLeR** [comment]
> Regarding the feature. It was not something I wanted to implement because in my opinion, it makes more sense to leave that job to Plex. And for example if you wanted to grab all Action movies in your library, I thought the better way to do it would be to create a smart playlist and then import that play list.

If you're looking to support multiple back-end media servers in the future, consolidating some more filtering functionality into dizqueTV does make sense.  Anything not incorporated would potentially require support/wiki to explain "how to" for multiple back-end media servers.

I like genre, but it comes with its challenges.  One is many movies are listed on TheMovieDB with up to five genres  Does the functionality consider any a match, or just the first one/two?  Star Wars is classified as an Adventure-Action-SciFi.  If the Action tag is applicable, then it could be right alongside Gladiator and 300.  It could get messy having movies as multiple genres.

**Comment by cbarbara** [comment]
> Just one thing I've noticed when testing it. When you click the + sign on the library, it will add the moveis that are inside the genres multiple times, because the library adder is also adding everything in each genre

thanks, just pushed the fix.

**Comment by vexorian** [comment]
Thank you for the feature.
